### DISCUSSION TITLE
destructuring should visit children of for stmt or not ?

### DISCUSSION THREAD

**Discussion by stormslowly** [swc-project/swc#9647] [General]
hi @kdy1 : 

in destructuring visitor

https://github.com/swc-project/swc/blob/88a2186ba419c98c73b997ca9ea90d7a8fd128e4/crates/swc_ecma_compat_es2015/src/destructuring.rs#L129-L140

after transform the first "level"  for of stmt,  it just skips visiting its children, so the nested for of stmt will be ignored, if we use destructuring separately .

```js
for (const [x,y] of arr1){
  for(const [a,b] of arr2 ){
     //... 
  }
}
```
is it by design or bug?

my workaround is using `for_of` transform before destructuring. or  destructuring is not supposed to used separately ?

**Comment by kdy1** [comment]
It‘s a definitely a bug

**Reply by stormslowly** [reply]
OK, let me fix that
